src/olympia/zadmin/tasks.py

```diff
@@ -587,14 +587,30 @@ def fetch_langpack(url, xpi, **kw):
             log.info('[@None] Created new "{0}" language pack, version {1}'
                      .format(xpi, data['version']))
 
+        # Set the category
+        for app in version.compatible_apps:
+            static_category = (
+                CATEGORIES[app.id][amo.ADDON_LPAPP]['general'])
+            category, _ = Category.objects.get_or_create(
+                id=static_category.id, defaults=static_category.__dict__)
+            AddonCategory.objects.get_or_create(addon=addon, category=category)
+        del addon.all_categories
+
+        # Add a license if there isn't one already
+        if not version.license:
+            license = License.objects.builtins().get(builtin=1)
+            version.update(license=license)
+
         file_ = version.files.get()
         if not is_beta:
             # Not `version.files.update`, because we need to trigger save
             # hooks.
             file_.update(status=amo.STATUS_PUBLIC)
         sign_file(file_, settings.SIGNING_SERVER)
 
-        addon.update_version()
+        addon.update(status=amo.STATUS_PUBLIC,
+                     summary=(addon.summary if addon.summary else addon.name))
```

Comment:
It's a fallback for potentially missing summary, yes.  `Addon.summary` is nullable, but a summary is one of the pieces of meta-data required to be 'complete' and eligible to be publicly available.
I'll add a test.